### ISSUE TITLE
:strftime_iso_kitchen directive (%r) discards previously applied date directives.

### ISSUE THREAD

**Issue by sparta-developers** [bitwalker/timex#446]
### Steps to reproduce

```
"July 24, 2018 5:34:04 PM PDT"
  |> Timex.parse!("%B %e, %Y %r %Z", :strftime)
  |> Timex.format!("%B %e, %Y %r %Z", :strftime)
```

The output of this is `"July 25, 2018 05:34:04 PM PST8PDT"`

### Description of issue

I would expect the conversion to remain July 24.

**Comment by sparta-developers** [comment]
We discovered this bug on version `3.3.0`, but reproduced it on master.

**Comment by ckhrysze** [comment]
The fix was merged and included in the 3.4.1 release, thank you!
